### PR TITLE
restrict usergroup data

### PR DESCRIPTION
**Pull request by phoenixAja**
### Description

restrict the information returned from usergroup api endpoint

#### Issue
[ch137500](https://app.clubhouse.io/genepi/story/137500)

### Test plan

changed test to expect new data

### CODE REVIEW

File: src/backend/aspen/app/views/tests/test_usergroup_views.py [hunk 1]
```diff
@@ -12,7 +12,12 @@ def test_usergroup_view_get(session, app, client):
     with client.session_transaction() as sess:
         sess["profile"] = {"name": user.name, "user_id": user.auth0_user_id}
     res = client.get("/api/usergroup")
-    expected = {"user": user.to_dict(), "group": group.to_dict()}
+    expected = {
+        "user": {
+            k: v for k, v in user.to_dict().items() if k in ["name", "agreed_to_tos"]
+        },
+        "group": {k: v for k, v in user.group.to_dict().items() if k in ["name"]},
+    }
```
Comment (ttung) [inline src/backend/aspen/app/views/tests/test_usergroup_views.py]: I would say that at this point, you might as well be explicit about what you're returning:

```
{"user": { "name": "xxx", "agreed_to_tos": "yyy"}, "group": {"name": "zzz"}}
```

**Review by ttung — CHANGES_REQUESTED** [review]

File: src/backend/aspen/app/views/tests/test_usergroup_views.py (hunk 1 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_usergroup_views.py]: i made some changes where i wrap this in a function, maybe it's a better approach

Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_usergroup_views.py]: also i tried doing:
`session.query(User.name, User.agreed_to_tos, Group.name).select_from(User).join(User.Group)` and 
`session.query(User).options(load_only(“agreed_to_tos”, “name”), joinedload(User.group).load_only(“name”))` to do the filtering in the query itself, but both methods had faults

File: src/backend/aspen/app/views/api_utils.py [hunk 2]
```diff
@@ -8,6 +8,13 @@
 from aspen.database.models.usergroup import User
 
 
+def filter_usergroup_dict(
+    unfiltered_dict: Mapping[str, Union[str, bool]],
+    fields_to_keep: Iterable[Union[str, bool]],
```
Comment (ttung) [inline src/backend/aspen/app/views/api_utils.py]:
```suggestion
    fields_to_keep: Collection[str],
```

File: src/backend/aspen/app/views/tests/test_usergroup_views.py [hunk 3]
```diff
@@ -12,7 +13,10 @@ def test_usergroup_view_get(session, app, client):
     with client.session_transaction() as sess:
         sess["profile"] = {"name": user.name, "user_id": user.auth0_user_id}
     res = client.get("/api/usergroup")
-    expected = {"user": user.to_dict(), "group": group.to_dict()}
+    expected = {
+        "user": filter_usergroup_dict(user.to_dict(), ["name", "agreed_to_tos"]),
+        "group": filter_usergroup_dict(user.group.to_dict(), ["name"]),
```
Comment (ttung) [inline src/backend/aspen/app/views/tests/test_usergroup_views.py]: These allowed fields should be constants (tuples?  frozensets?) in `src/backend/aspen/app/views/usergroup.py`.

**Review by ttung — CHANGES_REQUESTED** [review]
very close!

File: src/backend/aspen/app/views/tests/test_usergroup_views.py (hunk 3 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_usergroup_views.py]: added some tuples!

File: src/backend/aspen/app/views/api_utils.py (hunk 2 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/api_utils.py]: what's the main difference between iterable and collection? They seem the same to me

Comment (ttung) [reply src/backend/aspen/app/views/api_utils.py]: Collection supports the `in` operation.

**Review by ttung — APPROVED** [review]

File: src/backend/aspen/app/views/api_utils.py (hunk 2 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/api_utils.py]: oh interesting that makes sense

**Pull request merged**